query-connector/src/app/(pages)/auditLogs/page.tsx

```diff
@@ -0,0 +1,328 @@
+"use client";
+
+import { useState, useEffect, ChangeEvent, useMemo } from "react";
+import styles from "./auditLogs.module.scss";
+import classNames from "classnames";
+import SearchField from "@/app/ui/designSystem/searchField/SearchField";
+import Table from "@/app/ui/designSystem/table/Table";
+import {
+  Button,
+  Select,
+  Pagination,
+  DatePicker,
+} from "@trussworks/react-uswds";
+import WithAuth from "@/app/ui/components/withAuth/WithAuth";
+
+/**
+ * Client component for the Audit Logs page.
+ * @returns - The AuditLogs component.
+ */
+const AuditLogs: React.FC = () => {
+  const [search, setSearch] = useState("");
+  const [selectedName, setSelectedName] = useState("");
+  const [selectedAction, setSelectedAction] = useState("");
+  const [selectedDate, setSelectedDate] = useState<Date | null>(null);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [actionsPerPage, setActionsPerPage] = useState(10);
+
+  const logs = useMemo(() => {
+    const baseData = [
+      {
+        name: "Rocky Balboa",
+        action: "Created Report",
+        date: new Date("2025-03-10T14:30:00Z"),
+      },
+      {
+        name: "Apollo Creed",
+        action: "Edited Report",
+        date: new Date("2025-03-09T09:15:00Z"),
+      },
+      {
+        name: "Rocky Balboa",
+        action: "Deleted Entry",
+        date: new Date("2022-03-08T17:45:00Z"),
+      },
+      {
+        name: "Clubber Lang",
+        action: "Created Report",
+        date: new Date("2024-03-07T12:00:00Z"),
+      },
+      {
+        name: "Ivan Drago",
+        action: "Viewed Entry",
+        date: new Date("2025-03-06T22:10:00Z"),
+      },
+    ];
+
+    return Array.from({ length: 50 }, (_, index) =>
+      baseData.map((entry) => ({
+        ...entry,
+        date: new Date(entry.date.getTime() + index * 86400000),
+      })),
+    ).flat();
+  }, []);
+
+  const uniqueNames = useMemo(
+    () => Array.from(new Set(logs.map((log) => log.name))).sort(),
+    [logs],
+  );
+  const uniqueActions = useMemo(
+    () => Array.from(new Set(logs.map((log) => log.action))).sort(),
+    [logs],
+  );
+
+  const [dateError, setDateError] = useState("");
+  const minDate = useMemo(
+    () =>
+      logs.length > 0
+        ? new Date(Math.min(...logs.map((log) => log.date.getTime())))
+        : null,
+    [logs],
+  );
+  const maxDate = useMemo(
+    () =>
+      logs.length > 0
+        ? new Date(Math.max(...logs.map((log) => log.date.getTime())))
+        : null,
+    [logs],
+  );
+
+  useEffect(() => {
+    const dateInput = document.querySelector(
+      ".usa-date-picker__external-input",
```

Comment:
we may want to use a querySelector here that isn't a class / is more stable to CSS changes